File: 연속합.py
import sys
N = int(input())
array = list(map(int,input().split()))
maxValue = 0
dp = [-1000 for i in range(N) ] 
for i in range(N):
  for j in range(N-i):
    if i==0 :
      dp[j] = array[j]
    else :
      dp[j] = dp[j]+array[j+i]
  if i == 0:
    maxValue = max(dp)
  else :
    bin = max(dp)
    if maxValue < bin:
      maxValue = bin

print(maxValue)

File: 연속합.js
const fs = require("fs");
let input = fs.readFileSync('./input.txt').toString().split('\n');

const array = input[1].split(' ').map(el => +el);

const dp =[];
array.forEach(el=>{
  if(dp.length ===0){
    dp.push(el);
  }else{
    dp.push(Math.max(el, el + dp[dp.length-1]));
  }
});
console.log(dp);
const max = dp.reduce((el1, el2) => {
  console.log(el1, el2)
  return el1 < el2 ? el2 : el1
})

const solution = (max) =>{
  console.log(max);
}
solution(max);
